hack/make-rules/test-cmd-util.sh

```diff
@@ -2862,7 +2862,7 @@ run_deployment_tests() {
 
   ### Test kubectl create deployment should not fail validation
   # Pre-Condition: No deployment exists.
-  kube::test::get_object_assert deployment "{{range.items}}{{$id_field}}:{{end}}" ''
+  kube::test::wait_object_assert deployment "{{range.items}}{{$id_field}}:{{end}}" ''
```

Comment:
Why do we need this change? The behavior of `kubectl delete deployment` at line 2861 shouldn't change.

Here and several other occurrences.